pyface/action/field_action.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) 2019, Enthought, Inc.
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in enthought/LICENSE.txt and may be redistributed only
+# under the conditions described in the aforementioned license.  The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+# Thanks for using Enthought open source!
+
+from traits.api import Any, Constant, Dict, Str, Type
+
+from pyface.fields.i_field import IField
+from .action import Action
+
+
+class FieldAction(Action):
+    """ A widget action containing an IField
+
+    When the value in the field is changed, the `on_peform` method is called
+    with the new value as the argument.
+    """
+
+    #: This is a widget action.
+    style = Constant("widget")
+
+    #: The field to display.
+    field_type = Type(IField)
+
+    #: The field to display.
+    field_defaults = Dict(Str, Any)
+
+    def create_control(self, parent):
+        """ Called when creating a "widget" style action.
+
+        This constructs an IField-based control directly and binds changes to
+        the value to the `value_updated` method.
+
+        Parameters
+        ----------
+        parent : toolkit control
+            The toolkit control, usually a toolbar.
+
+        Returns
+        -------
+        control : toolkit control
+            A toolkit control or None.
+        """
+        field = self.field_type(parent=parent, **self.field_defaults)
+        field._create()
+        field.on_trait_change(self.value_updated, 'value')
+        field.control._field = field
```

Comment:
Are you making a reference to `field` here because no other object will get a reference? (because you're returning `field.control`)

Reply:
This is a hack to get a long-lived reference to `field` so that it isn't GC'd during the lifetime of the underlying toolkit control.  This is done in part because `Action`s don't know whether they have a 1-to-1 or 1-to-many relationship with the concrete toolbar controls... not how I would code all of this from scratch.

I do a similar thing with the `UI` instance for the TraitsUI code for the same reason.

Reply:
I suspected that was what was going on.